substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -243,14 +245,28 @@ pub fn migrate_farms<T: Config>() -> frame_support::weights::Weight {
             }
         }
 
-        let replaced_farm_name = farm::replace_farm_name_spaces_with_underscores(&farm.name);
-        let name = match <T as Config>::FarmName::try_from(replaced_farm_name) {
-            Ok(n) => n,
-            Err(_) => {
-                info!("invalid farm name, skipping updating farm {:?} ...", k);
-                return None;
-            }
-        };
+        let mut farm_name = farm.name;
+        let truncated = farm_name.len() > <T as Config>::MaxFarmNameLength::get() as usize;
+        if truncated {
+            info!("farm name length exceeded, truncating farm name");
+            farm_name.truncate(<T as Config>::MaxFarmNameLength::get() as usize);
+        }
+
+        let mut replaced_farm_name = farm::replace_farm_name_invalid_characters(&farm_name);
+        let name = <T as Config>::FarmName::try_from(replaced_farm_name.clone()).unwrap_or({
+            replaced_farm_name = b"change_me_".to_vec();
+            let first = ((generated_farm_names / 26) + b'a' as u32) as u8;
+            let second = ((generated_farm_names % 26) + b'a' as u32) as u8;
+            replaced_farm_name.append(&mut [first, second].to_vec());
```

Comment:
```suggestion
            replaced_farm_name.extend_from_slice(&[first, second]);
```

Using append should only be done when the existing data is already in a vector, and expensive to copy, as then it can just move the data. Append also requires a Vec, which means you need to allocate and copy the data from the stack into the heap in the vector, which will be worse performance, as you allocate + move twice, whereas this will just copy once. Given the data is bytes, a copy is is as fast as moving.